semantic_conventions/trace/messaging.yaml

```
@@ -133,6 +135,7 @@ groups:
             If the key is `null`, the attribute MUST NOT be set.
           note: >
             If the key type is not string, it's string representation has to be supplied for the attribute.
+            If the key can not be represented in string form, don't include its value.
```

I think you can find a string representation for everything if you try. So maybe we should say

```suggestion
            If the key has no unambiguous, canonical string form, don't include its value.
```